src/Plugin/EntityBrowser/Widget/View.php

```diff
@@ -199,11 +165,11 @@ public static function processCheckbox(&$element, FormStateInterface $form_state
    * {@inheritdoc}
    */
   public function submit(array &$element, array &$form, FormStateInterface $form_state) {
-    $selected_rows = array_keys(array_filter($form_state->getValue('entity_browser_select')));
+    $selected_rows = array_values(array_filter($form_state->getUserInput()['entity_browser_select']));
     $entities = [];
-    $ids = $form_state->get('view_widget_rows');
     foreach ($selected_rows as $row) {
-      $entities[] = $this->entityManager->getStorage($ids[$row]['type'])->load($ids[$row]['id']);
+      list($type, $id) = explode(':', Html::escape($row));
```

Comment:
Yeah it only replicates what the FAPI does with things like `#options`, `#default_value`, etc. I'll update the validation a bit to make sure we're dealing with a valid string and that the type exists.